src/parser.rs

```diff
@@ -211,6 +213,48 @@ where
         self.parse_program()
     }
 
```

Comment:
longer comment why this thing exists and why this design decision was made. Primarily for me, but Andreas and Sebastian will have the same question when the review our project.